src/utils/yaml_node_utils.cpp

```diff
@@ -209,15 +209,11 @@ YAML::Node mp::utils::make_cloud_init_network_config(
     const auto default_interface = interface_details{default_mac_addr};
     network_data["ethernets"][default_interface.name] = default_interface;
 
-    auto extra_idx = kExtraInterfaceIndexStart;
-    // FIXME: C++20 range-for init statement:
-    // https://en.cppreference.com/w/cpp/language/range-for.html
-    // C++23: https://en.cppreference.com/w/cpp/ranges/enumerate_view.html
-    for (const auto& extra : extra_interfaces)
+    // TODO@C++23: https://en.cppreference.com/w/cpp/ranges/enumerate_view.html
+    for (auto extra_idx = kExtraInterfaceIndexStart; const auto& extra : extra_interfaces)
     {
         if (!extra.auto_mode)
             continue;
-        // FIXME: C++20 (designated initializers)
         const auto extra_interface = interface_details{/*mac_addr=*/extra.mac_address,
                                                        /*interface_idx=*/extra_idx,
                                                        /*optional=*/true};
```

Comment:
`interface_details` has a constructor, so we can't use designated initialization with it.